test/gtest/ucp/test_ucp_peer_failure.cc

```diff
@@ -217,31 +232,57 @@ void test_ucp_peer_failure::fail_receiver() {
          * message that are expected here, since we closed receiver w/o
          * reading the messages that were potentially received */
         scoped_log_handler slh(warn_unreleased_rdesc_handler);
+
+        cleanup_rndv_descs();
         failing_receiver().cleanup();
     }
 }
 
-void test_ucp_peer_failure::smoke_test(bool stable_pair)
+std::pair<ucs_status_t, ucs_status_t>
+test_ucp_peer_failure::smoke_test(bool stable_pair)
 {
     ucp_ep_h send_ep = stable_pair ? stable_sender() : failing_sender();
     size_t am_count  = m_am_rx_count;
+    ucp_request_param_t req_param{};
+    std::pair<ucs_status_t, ucs_status_t> result;
 
     // Send and wait for completion
     void *sreq = send_nb(send_ep, stable_pair ? m_stable_rkey : m_failing_rkey);
-    request_wait(sreq);
 
     if (get_variant_value() & TEST_AM) {
         // Wait for active message to be received
-        while (m_am_rx_count < am_count) {
+        while (m_am_rx_count == am_count) {
             progress();
         }
-    } else if (get_variant_value() & TEST_RMA) {
+
+        auto &recv_entity = stable_pair ? stable_receiver() :
+                                          failing_receiver();
+        while (!m_rndv_descs.empty()) {
+            void* rreq    = ucp_am_recv_data_nbx(recv_entity.worker(),
+                                                 m_rndv_descs.front(),
+                                                 &m_rbuf[0], m_rbuf.size(),
+                                                 &req_param);
+            result.second = request_wait(rreq);
+            m_rndv_descs.pop();
+        }
+    } else if (get_variant_value() & TEST_TAG) {
+        void* rreq    = ucp_tag_recv_nbx(receiver().worker(), &m_rbuf[0],
+                                         m_rbuf.size(), 0, 0, &req_param);
+        result.second = request_wait(rreq);
+    }
+
+    result.first = request_wait(sreq);
+
+    if (get_variant_value() & TEST_RMA) {
```

Comment:
can we make it as 'else if' right after line 272?
we can move 'result.first = request_wait(sreq);' after it, since it's just waiting, so order is not important